src/avatar.cls.php

```diff
@@ -270,7 +240,7 @@ private function _generate( $url ) {
 		self::save_summary();
 
 		// Generate
-		if ( ! self::has_cache() ) {
+		if ( ! $this->has_cache_folder( 'avatar', $this->cls('Purge')->_is_subsite_purge() ? get_current_blog_id() : false ) ) {
```

Comment:
This is to generate avatar, shouldn't use `_is_subsite_purge` function to detect. As you are using the blog_id detection in https://github.com/litespeedtech/lscache_wp/pull/386/files#diff-9456edaa82557d2a23d04e2abe2b665a0a8c31255da28f97fbe5d30ca21a07daR383 too, maybe you can extract them to a function.

Reply:
got it!

Reply:
I did this 
`// Generate
		if ( ! $this->has_cache_folder( 'avatar' ) ) {
			$this->_mkdir();
		}`
and for Root class
`public function has_cache_folder( $type ) {
		$subsite_id = is_multisite() && ! is_network_admin() ? get_current_blog_id() : '';

		if ( file_exists( LITESPEED_STATIC_DIR . '/' . $type . '/' . $subsite_id ) ) {
			return true;
		}
		return false;
	}`